infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -46,13 +45,18 @@
 import static org.corfudb.infrastructure.logreplication.LogReplicationConfig.REGISTRY_TABLE_ID;
 
 /**
- * This class represents the entity responsible of writing streams' snapshots into the standby cluster DB.
+ * This class represents the entity responsible for writing streams' snapshots into the standby cluster DB.
  *
- * Snapshot sync is the process of transferring a snapshot of the DB, for this reason, data is temporarily applied
- * to shadow streams in an effort to avoid inconsistent states. Once all the data is received, the shadow streams
- * are applied into the actual streams.
+ * Snapshot sync is the process of transferring a snapshot of the DB. We create a shadow stream per stream to
+ * replicate. A shadow stream aims to accumulate updates temporarily while the (full) snapshot sync completes.
+ * Shadow streams aim to avoid inconsistent states while data is still being transferred from active to standby
+ * Once all the data is received, the shadow streams are applied into the actual streams.
+ *
+ * We currently, wait for snapshot sync to complete before applying data in shadow streams to the actual streams,
+ * this means that there is still a window of inconsistency as apply is not atomic, but at least inconsistency is
```

Comment:
inconsistency is not because data is first applied to shadow streams.  It is because all data from shadow to actual streams may not be applied in a single transaction if the size exceeds the Corfu transaction size limit.  Can you reword the comment accordingly?